static/js/components/tiles/histogram_tile.tsx

```diff
@@ -18,177 +18,200 @@
-    </ChartTileContainer>
-  );
+  // Fill in months between start and end dates
+  const months = new Array<string>();
+  for (
+    let date = new Date(startDate);
+    getMonthString(date) <= getMonthString(new Date(endDate));
+    date.setMonth(date.getMonth() + 1)
+  ) {
+    months.push(getMonthString(new Date(date)));
+  }
+  return months;
 }
 
 /**
- * Gets the series of observations to plot
+ * Helper function to bin data by month
  */
-function fetchData(
-  props: HistogramTilePropType,
-  setRawData: (data: SeriesApiResponse) => void
-): void {
-  const statVars = [];
-  for (const spec of props.statVarSpec) {
-    statVars.push(spec.statVar);
-    if (spec.denom) {
-      statVars.push(spec.denom);
+function binDataByMonth(
+  disasterEventPoints: DisasterEventPoint[],
+  dateSetting: string
+): DataPoint[] {
+  if (_.isEmpty(disasterEventPoints)) {
+    return [];
+  }
+
+  // Track YYYY-MM -> number of events
+  const bins = new Map<string, number>();
+
+  // Initialize all bins at zero
+  const monthsToPlot = getMonthsArray(dateSetting, disasterEventPoints);
+  for (const month of monthsToPlot) {
+    bins.set(month, 0);
+  }
+
+  for (const event of disasterEventPoints) {
+    // Get start time in YYYY-MM
+    const eventMonth = event.startDate.slice(0, "YYYY-MM".length);
+
+    // Increment count in corresponding bin
+    if (bins.has(eventMonth)) {
+      bins.set(eventMonth, bins.get(eventMonth) + 1);
     }
   }
-  axios
-    .get("/api/observations/series/binned", {
-      // Fetch both numerator stat vars and denominator stat vars
-      params: {
-        variables: statVars,
-        entities: [props.place.dcid],
-      },
-      paramsSerializer: stringifyFn,
-    })
-    .then((resp) => {
-      setRawData(resp.data);
-    })
-    .catch((error) => {
-      console.log(error);
-      setRawData(null);
-    });
-}
 
-/**
- * Process and set histogram data
- */
-function processData(
-  props: HistogramTilePropType,
-  rawData: SeriesApiResponse,
-  setHistogramData: (data: HistogramData) => void
-): void {
-  const histogramData = rawToChart(rawData, props);
-  setHistogramData(histogramData);
+  // Format binned data into DataPoint[]
+  const histogramData = new Array<DataPoint>();
+  bins.forEach((value, label) => {
+    histogramData.push({ label: label, value: value });
+  });
+
+  return histogramData;
 }
 
 /**
- * Renders the histogram
+ * Helper function to determine if data exists to display.
  */
-function draw(props: HistogramTilePropType, histogramData: DataPoint[]): void {
-  const elem = document.getElementById(props.id);
-  // TODO: Remove all cases of setting innerHTML directly.
-  elem.innerHTML = "";
-  drawHistogram(
-    props.id,
-    elem.offsetWidth,
-    props.svgChartHeight,
-    histogramData,
-    props.statVarSpec[0].unit
-  );
+function shouldShowHistogram(histogramData: DataPoint[]): boolean {
+  return histogramData && !_.isEmpty(histogramData);
```

Comment:
nit: _.isEmpty will check for nulls and undefined too, so really only need to keep the ~_.isEmpty check here